app/views/includes/sanity.scala.txt

```diff
@@ -1,7 +1,23 @@
 # Sanity checks
 
+echo "Validating installation location..."
+if [ -e "$SDKMAN_DIR" ] && [ ! -d "$SDKMAN_DIR" ]; then
+	echo "Cannot install to requested location."
+	echo ""
+	echo "======================================================================================================"
+	echo " A file of the same name as the installation directory exists."
```

Comment:
I really don't see why we should check for the presence of a file here. This seems like a very odd situation to be in and perhaps something that you've gotten yourself into by accidentally touching a file called `~/.sdkman`. This seems highly unlikely to ever happen out in the wild and feels over-defensive to me.

Reply:
I understand the concern, just offering it up while I was in making enhancements.  The installer makes a lot of assumptions about the availability of the installation directory and throws all sorts of errors if there is a file in the way.  It's obviously your call, but what's the harm in better testing that is simple?